app/resources/scanned_resources/scanned_resources_controller.rb

```diff
@@ -23,10 +23,14 @@ def after_create_success(obj, change_set)
 
   def change_set_class
     if params[:change_set].present? || (resource_params && resource_params[:change_set].present?)
-      DynamicChangeSet.class_from_param(params[:change_set] || resource_params[:change_set])
+      change_set_name = params[:change_set] || resource_params[:change_set]
+      DynamicChangeSet.class_from_param(change_set_name)
     else
       DynamicChangeSet
     end
+  rescue NameError
```

Comment:
If the problem is that this error occurs in several different contexts, maybe it would be better to handle it in `DynamicChangeSet` instead of in the controller?